### PR TITLE
Adding SDK for Peering

### PR DESCRIPTION
**Pull request by swetha-anantharajaiah**
Adding SDK for Peering

### CODE REVIEW

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
This PR cannot be accepted.
Please work with @dsgouda who can help you to make sure csproj file is accurate and can be built for multiple targets.

File: src/SDKs/Peering/Management.Peering/Microsoft.Azure.Management.Peering.csproj
Comment (dsgouda) [inline src/SDKs/Peering/Management.Peering/Microsoft.Azure.Management.Peering.csproj]: Versioning needs to start from 0.8.0-preview

File: src/SDKs/Peering/Management.Peering/Microsoft.Azure.Management.Peering.csproj
Comment (dsgouda) [inline src/SDKs/Peering/Management.Peering/Microsoft.Azure.Management.Peering.csproj]: Please remove lines 15 through 40

File: src/SDKs/Peering/Management.Peering/Microsoft.Azure.Management.Peering.csproj
Comment (dsgouda) [inline src/SDKs/Peering/Management.Peering/Microsoft.Azure.Management.Peering.csproj]: Please remove this ItemGroup.

File: src/SDKs/Peering/Management.Peering/Microsoft.Azure.Management.Peering.csproj
Comment (dsgouda) [inline src/SDKs/Peering/Management.Peering/Microsoft.Azure.Management.Peering.csproj]: Please take a look at a csproj [here](https://github.com/Azure/azure-sdk-for-net/blob/master/src/SDKs/Blueprint/Management.Blueprint/Microsoft.Azure.Management.Blueprint.csproj) and remove any additional propertygroups/itemgroups

File: src/SDKs/Peering/Management.Peering/Microsoft.Azure.Management.Peering.csproj
Comment (dsgouda) [inline src/SDKs/Peering/Management.Peering/Microsoft.Azure.Management.Peering.csproj]: Please remove `Description` use `PackageReleaseNotes` instead

File: src/SDKs/Peering/Management.Peering.Tests/Peering.Tests.csproj
Comment (dsgouda) [inline src/SDKs/Peering/Management.Peering.Tests/Peering.Tests.csproj]: Please remove this `PropertyGroup`. Refer this and remove any additional `PropertyGroup`/`ItemGroup`

**Review by dsgouda — CHANGES_REQUESTED** [review]
Please refer to sample csproj files and make changes

**Comment by dsgouda** [comment]
@swetha1microsoft please link the REST spec PR

**Comment by swetha1microsoft** [comment]
> @swetha1microsoft please link the REST spec PR

**Comment by swetha1microsoft** [comment]
> @swetha1microsoft please link the REST spec PR

https://github.com/Azure/azure-rest-api-specs/pull/5343

File: src/SDKs/Peering/Management.Peering.Tests/Peering.Tests.csproj
Comment (dsgouda) [inline src/SDKs/Peering/Management.Peering.Tests/Peering.Tests.csproj]: Test csproj must be versioned to `1.0.0` and never modified

File: src/SDKs/Peering/Management.Peering/Properties/AssemblyInfo.cs
Comment (dsgouda) [inline src/SDKs/Peering/Management.Peering/Properties/AssemblyInfo.cs]: Rename this to Peering

File: src/SDKs/Peering/Management.Peering/generated/EdgeRpClient.cs
Comment (dsgouda) [inline src/SDKs/Peering/Management.Peering/generated/EdgeRpClient.cs]: Was the `generated` directory auto-generated

**Review by dsgouda — CHANGES_REQUESTED** [review]
Please delete this src/SDKs/Peering/Management.Peering/MSSharedLibKey.snk

File: src/SDKs/Peering/Management.Peering/generated/EdgeRpClient.cs
Comment (swetha1microsoft) [reply src/SDKs/Peering/Management.Peering/generated/EdgeRpClient.cs]: yes

File: src/SDKs/Peering/Management.Peering/Properties/AssemblyInfo.cs
Comment (swetha1microsoft) [reply src/SDKs/Peering/Management.Peering/Properties/AssemblyInfo.cs]: done

File: src/SDKs/Peering/Management.Peering.Tests/Peering.Tests.csproj
Comment (swetha1microsoft) [reply src/SDKs/Peering/Management.Peering.Tests/Peering.Tests.csproj]: done

**Comment by dsgouda** [comment]
@shahabhijeet is this PR still needed?

**Comment by swetha1microsoft** [comment]
> @shahabhijeet is this PR still needed?

We are getting the REST Spec PR reviewed and resolving the comments, once the Spec PR is completed, we will need sign off on the SDK PR.

**Comment by dsgouda** [comment]
> 
> 
> > @shahabhijeet is this PR still needed?
> 
> We are getting the REST Spec PR reviewed and resolving the comments, once the Spec PR is completed, we will need sign off on the SDK PR.

Sounds good, please ensure the PR is updated with latest changes from upstream

**Comment by rileymckenna** [comment]
Updated to the latest Swagger per the Spec PR. 
Please see the [Rest Spec](https://github.com/Azure/azure-rest-api-specs/pull/5343) @dsgouda @shahabhijeet we are awaiting the merge.
@swetha1microsoft

**Review by dsgouda — CHANGES_REQUESTED** [review]
Please create a generate.ps1 script similar to [this](https://github.com/Azure/azure-sdk-for-net/blob/master/src/SDKs/Billing/Management.Billing/generate.ps1) and regenerate the code when the Rest spec is merged.

File: src/SDKs/Peering/Management.Peering.Tests/Peering.Tests.csproj
Comment (dsgouda) [inline src/SDKs/Peering/Management.Peering.Tests/Peering.Tests.csproj]: Please rename the directory `Management.Peering.Tests` to `Peering.Tests`

File: src/SDKs/Peering/Management.Peering/Properties/AssemblyInfo.cs
Comment (dsgouda) [inline src/SDKs/Peering/Management.Peering/Properties/AssemblyInfo.cs]: nit: Undo these changes

**Review by dsgouda — CHANGES_REQUESTED** [review]
Couple of minor comments apart, looks great. Approving in principle, will approve once the Rest spec is merged.

File: src/SDKs/Peering/Management.Peering/Properties/AssemblyInfo.cs
Comment (rileymckenna) [inline src/SDKs/Peering/Management.Peering/Properties/AssemblyInfo.cs]: i am not sure what the change was. They seem to look the same.

File: src/SDKs/Peering/Management.Peering.Tests/Peering.Tests.csproj
Comment (rileymckenna) [reply src/SDKs/Peering/Management.Peering.Tests/Peering.Tests.csproj]: in the latest the name was changed to Peering.Tests.

**Comment by dsgouda** [comment]
Please fix the failing tests too

**Comment by rileymckenna** [comment]
@dsgouda [Arm Signed off](https://github.com/Azure/azure-rest-api-specs/pull/5343)

**Comment by rileymckenna** [comment]
why does the proj not build?
Error: c:\workspace\VSUpgradeForTest\tools\SdkBuildTools\targets\common.targets(347,5): error MSB3073: The command "dotnet test -f netcoreapp2.0 c:\workspace\VSUpgradeForTest\src\SDKs\Peering\Management.Peering.Tests\Peering.Tests.csproj -l trx;LogFileName=c:\workspace\VSUpgradeForTest\TestResults\netcoreapp2.0\Peering.Tests.trx /nodeReuse:false" exited with code 1. [c:\workspace\VSUpgradeForTest\build.proj]

**Comment by dsgouda** [comment]
@rileymckenna are you able to locally run the tests?

**Comment by rileymckenna** [comment]
@dsgouda i am passing tests now. i figured out the record

**Comment by rileymckenna** [comment]
@dsgouda  fixed the test and now its passed everything

**Review by dsgouda — CHANGES_REQUESTED** [review]
Please regenerate the code using [generate.ps1](https://github.com/Azure/azure-sdk-for-net/blob/4cc472c3a1a9e8c560ecc3d333a1f1d96cc8e36e/src/SDKs/Peering/Management.Peering/generate.ps1) and commit all changes generated.
Everything looks great otherwise

**Review by dsgouda — APPROVED** [review]
LGTM

**Comment by dsgouda** [comment]
Will merge once CIs pass

**Pull request merged**
